app/components/post_textbox/post_textbox.js

```diff
@@ -85,15 +97,23 @@ export default class PostTextbox extends PureComponent {
             top: 0,
             value: props.value,
             showFileMaxWarning: false,
+            keyboardState: 'hidden',
         };
     }
 
     componentDidMount() {
         const event = this.props.rootId ? INSERT_TO_COMMENT : INSERT_TO_DRAFT;
         EventEmitter.on(event, this.handleInsertTextToDraft);
+        Keyboard.addListener('keyboardDidShow', this.onKeyboardShow);
+        Keyboard.addListener('keyboardDidHide', this.onKeyboardHide);
+
         if (Platform.OS === 'android') {
-            Keyboard.addListener('keyboardDidHide', this.handleAndroidKeyboard);
             BackHandler.addEventListener('hardwareBackPress', this.handleAndroidBack);
+            this.listenerId = mattermostManaged.addEventListener('hardwareEnter', this.handleHardwareEnterKey);
+        } else if (Platform.OS === 'ios') {
+            const {EventEmitModule} = NativeModules;
+            this.eventEmitter = new NativeEventEmitter(EventEmitModule);
+            this.eventEmitter.addListener('handleIosEnter', this.handleHardwareEnterKey);
```

Comment:
Do we need a new native module just for this? is there another way to accomplish the same thing? Also try and keep both platforms consistent for example I don't get why the event is called different, this should be `hardwareEnter` the same as Android.